sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/FileRefDataObject.scala

```diff
@@ -143,12 +152,25 @@ private[smartdatalake] trait FileRefDataObject extends FileDataObject {
     }
   }
 
+  /**
+   * Define recommended number of files to be read or written in parallel for this DataObject.
+   * Actions using this DataObject are not forced to respect the parallelism given, but they can use the information to better control parallel operations.
+   * When set to None there is no special recommendation for parallelism.
+   * Note: this was implemented for FileTransferAction working with SFtpFileRefDataObject, to make use of SFtpFileRefConnection.maxParallelConnections.
+   */
+  def recommendedParallelism: Option[Int] = None
 
   /**
    * Delete all data. This is used to implement SaveMode.Overwrite.
    */
   def deleteAll(implicit context: ActionPipelineContext): Unit = throw new RuntimeException(s"($id) deleteAll not implemented")
 
+  /**
+   * Create directories if not existing.
+   * If no implementation is given, it is assumed that directories will be created on the file when writing a file.
```

Comment:
Does this mean that it is expected that the folder in which the file should be written already exists ?
Or is it expected that when writing the file, the folder is created within the same call ?

Reply:
This should be "will be created on-the-fly when writing a file." 
I'll fix that :-)